### PR TITLE
mgmt azure-json, fix flattened properties in parent class

### PR DESCRIPTION
**Pull request by XiaofeiCao**
- fix https://github.com/Azure/autorest.java/issues/2742
- Changes in this PR
  - Fixed flattened properties in parent class(code: https://github.com/Azure/autorest.java/pull/2817/commits/3bc20e231605f4b7382292ca2f3a1edb5adc242b)
    - Duplicated flattened property in child class
    - Override getter
    - Override setter
    - in `fromJson`, set shadowed flattened property, instead of the package private setter
  - Fixed duplicate field properties when child class defines properties that shadow the parent properties with the same names. (code: https://github.com/Azure/autorest.java/pull/2817/commits/a224119af35ac54677427267e4bd027c22463599)

Regened compute and network, `mvn verify` passed for both.

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java [hunk 1]
```diff
@@ -412,27 +428,30 @@ protected void addSerializationImports(Set<String> imports, ClientModel model, J
...
                     // parent property doesn't have setter
                     if (!ClientModelUtil.hasSetter(parentProperty, settings)) {
                         return false;
                     }
                     // child does not contain property that shadow this parent property
                     return !modelPropertyNames.contains(parentProperty.getName());
-                }
-            ).collect(Collectors.toList());
+                })
+                .collect(Collectors.toList());
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java:454]: Just format change?

**Review by weidongxu-microsoft — APPROVED** [review]

File: typespec-tests/tspconfig.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ options:
     generate-tests: true
     examples-directory: "{project-root}/tsp/examples"
     flavor: "Azure"
+    stream-style-serialization: true
```
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply typespec-tests/tspconfig.yaml:11]: yeah, let me remove that

File: javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java [hunk 3]
```diff
@@ -550,33 +550,24 @@ public static List<ClientModelProperty> getParentConstructorProperties(ClientMod
...
-    // Only applies to mgmt, no effect on vanilla or DPG.
-    private static boolean isPrivateAccess(ClientModelPropertyAccess property) {
-        boolean privateAccess = false;
-        // ClientModelPropertyReference never refers to a private access property, so only check ClientModelProperty here.
-        if (property instanceof ClientModelProperty) {
-            privateAccess = ((ClientModelProperty) property).getClientFlatten();
-        }
-        return privateAccess;
+    public static boolean needsPublicSetter(ClientModelPropertyAccess property, JavaSettings settings) {
+        return !isReadOnlyOrInConstructor(property, settings) && !isFlattenedProperty(property);
```
Comment (XiaofeiCao) [inline javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java:560]: Refactor to make this method's purpose more clear.

File: javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java (hunk 1 above)
Comment (XiaofeiCao) [reply javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java]: Yeah, changed back..

**Review by alzimmermsft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
You may do a follow-up PR to change the default (for Swagger) to true, and see how it works for existing test and the generation action https://github.com/Azure/autorest.java/actions/workflows/fluent-test-gen-all.yml

**Pull request merged**